src/transformStateWithClones.js

```diff
@@ -64,7 +64,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const statesArr = [];
+  const referenceArr = { ...state };
+
+  for (let i = 0; i < transforms.length; ++i) {
+    const { operation, properties } = transforms[i];
+
+    switch (operation) {
+      case 'addProperties' : {
+        for (const key in properties) {
+          referenceArr[key] = properties[key];
+        }
+
+        statesArr[statesArr.length] = Object.assign({}, referenceArr);
+        break;
+      }
+
+      case 'clear' : {
+        for (const key in referenceArr) {
+          delete referenceArr[key];
+        }
+
+        statesArr.push(Object.assign({}, referenceArr));
```

Comment:
чому додав `push` тільки в цьому місці, а в інших ні?

Reply:
ти можеш використовувати новий синтаксис
```suggestion
        statesArr.push({ ...referenceArr });
```

Reply:
почитай як працює spread operator